### PR TITLE
git-merge-repo

### PR DESCRIPTION
**Pull request by imsky**
This command merges two repo histories. There's an optional `--squash` flag and an optional "flat merge" mode - if the incoming repo's directory is set to `.`, then a squash-like commit is added for the files added to the current directory.

### CODE REVIEW

File: bin/git-merge-repo [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+branch=$2
+prefix=$3
+flat=0
+
+if test $prefix = "."; then
+	prefix=$(mktemp -u 'git-merge-repo.XXXXXXX')
+	flat=1
+fi
+
+git subtree add -P $prefix $repo $branch
```
Comment (hemanth) [inline bin/git-merge-repo]: Would you mind quoting the variables?

**Comment by hemanth** [comment]
Please squash the commits, once merged, don't forget to update the wiki.

**Comment by imsky** [comment]
done and done

**Comment by hemanth** [comment]
Sweet! :+1:

**Pull request merged**